### ISSUE TITLE
unable to add adt assay

### ISSUE THREAD

**Issue by yw2875** [satijalab/seurat#5412]
Hi there,

First of all, thank you for the wonderful package!

I am trying to add an adt assay on to a Seurat Object. My data is in h5 file, and I am only 1 feature barcoding in the CITE-seq. Here is my code:
Ctl.data <- Read10X_h5("A:/R/.../IgG_filtered_feature_bc_matrix.h5", use.names = TRUE, unique.features = TRUE)
Ctl.rna <- Ctl.data[["Gene Expression"]]
Ctl <- CreateSeuratObject(counts = Ctl.rna, project = "IgG")
Ctl.adt <- Ctl.data[["Antibody Capture"]]
Ctl[["ADT"]] <- CreateAssayObject(counts = Ctl.adt)

However, the feedback was:
"Error in CreateAssayObject(counts = Ctl.adt) : 
  No cell names (colnames) names present in the input matrix"

I read issue#1650 and realized that is could be that there is no cell names in the matrix read. The Ctl.adt contains Named Numeric vectors, and when I use:
names(Ctl.adt)
I could see the cell names like "AGGGAGTAGGATGGTC-1". However, somehow the CreatAssayObject could not read the names. 

I tried to save the Ctl.adt as csv and load in again, but the columns seemed unmatched. 

And I tried to use row.names to call it out, but unfortunately in my dataset, since there is only 1 Ab, people who helped me to do the annotation and alignment through CellRanger seemed not to put the name in so the code:
rownames(Ctl.data$`Antibody Capture`)
came back as "NULL"

I am wondering if there is a way around this or the better way for me to fix this is re-do the annotation and alignment and add in a row name.

Thank yo so much!

Best regards,
Flora

[issue #1650](https://github.com/satijalab/seurat/issues/1650)

**Comment by mhkowalski** [comment]
It appears that you need to set the colnames on the `Ctl.adt` object. You might have to convert this from a vector into a matrix...can you please try the following and confirm that the colnames are set correctly?
```
t <- as.matrix(Ctl.adt, nrow = 1)
colnames(t)  <- names(Ctl.adt)
colnames(t)
Ctl[["ADT"]] <- CreateAssayObject(counts = t)
```

**Comment by yw2875** [comment]
Hi,

Thanks for the kind reply!

When I run

t <- as.matrix(Ctl.adt, nrow = 1)
colnames(t)  <- names(Ctl.adt)

It returned as:

 "Error in dimnames(x) <- dn :

  length of 'dimnames' [2] not equal to array extent"

I am assuming the problem lies in the fact that the column name of
Ctl.adt at this point is "Null" from the h5 file. And I am currently
trying to load the data directly from the matrix+features+barcode
instead of h5 to see if I could dodge the problem.....

But please do let me know if you have any idea of what is going on!

Thank you so much!

**Comment by mhkowalski** [comment]
Ok, I think I do not understand the structure of your data is, that you had indicated that ```names(Ctl.adt)``` gave the cell barcodes. You need the input to ``` CreateAssayObject()``` to be 1 row (for your 1 Ab) and n columns (where n is the number of cells). Furthermore, the column names must correspond to the barcodes.

I hope this helps and best of luck!

**Comment by yw2875** [comment]
Sorry for the confusion. The Ctl.adt is supposed to give the antibody
information as an add-on assay as you indicated, but it need the cell name
to match to the same cell for the gene expression information. However,
when I try to add it on, the error was that the column names cannot be
found. When I save it as csv, it looks like the csv I attached. Maybe the
error lies in that the cell names are defaultly set as row names instead of
column names? I tried to invert the column and the row, but it did not seem
to work.

I guess right now I do not know where the problem is....
